x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_alerts_filter_timeframe.tsx

```diff
@@ -0,0 +1,221 @@
+    days: [],
+    timezone,
+    hours: {
+      start: '00:00',
+      end: '24:00',
+    },
+  };
+  return useState<AlertsFilterTimeframe>(initialTimeframe || DEFAULT_TIMEFRAME);
+};
+const useTimeFormat = () => {
+  const dateFormatScaled: Array<[string, string]> = useUiSetting('dateFormat:scaled') ?? [
+    ['PT1M', 'HH:mm'],
+  ];
+  const [, PT1M] = dateFormatScaled.find(([key]) => key === 'PT1M') ?? ['', 'HH:mm'];
+  return PT1M;
+};
+
+export const ActionAlertsFilterTimeframe: React.FC<ActionAlertsFilterTimeframeProps> = ({
+  state,
+  onChange,
+}) => {
+  const timeFormat = useTimeFormat();
+  const [timeframe, setTimeframe] = useTimeframe(state);
+  const [selectedTimezone, setSelectedTimezone] = useState([{ label: timeframe.timezone }]);
+
+  const timeframeEnabled = useMemo(() => Boolean(state), [state]);
+
+  const weekdayOptions = useSortedWeekdayOptions();
+
+  useEffect(() => {
+    const nextState = timeframeEnabled ? timeframe : null;
+    if (!deepEqual(state, nextState)) onChange(nextState);
+  }, [timeframeEnabled, timeframe, state, onChange]);
+
+  const toggleTimeframe = useCallback(
+    () => onChange(state ? null : timeframe),
+    [state, timeframe, onChange]
+  );
+  const updateTimeframe = useCallback(
+    (update: Partial<AlertsFilterTimeframe>) => {
+      setTimeframe({
+        ...timeframe,
+        ...update,
+      });
+    },
+    [timeframe, setTimeframe]
+  );
+
+  const onChangeHours = useCallback(
+    (startOrEnd: 'start' | 'end') => (date: Moment) => {
+      updateTimeframe({
+        hours: { ...timeframe.hours, [startOrEnd]: date.format('HH:mm') },
+      });
+    },
+    [updateTimeframe, timeframe]
+  );
+
+  const onToggleWeekday = useCallback(
+    (id: string) => {
+      if (!timeframe) return;
+      const day = Number(id) as IsoWeekday;
+      const previouslyHasDay = timeframe.days.includes(day);
+      const newDays = previouslyHasDay
+        ? timeframe.days.filter((d) => d !== day)
+        : [...timeframe.days, day];
+      updateTimeframe({ days: newDays });
+    },
+    [timeframe, updateTimeframe]
+  );
+  const selectedWeekdays = useMemo(
+    () =>
+      ISO_WEEKDAYS.reduce(
+        (result, day) => ({ ...result, [day]: timeframe.days.includes(day) }),
+        {}
+      ),
+    [timeframe]
+  );
+
+  const onChangeTimezone = useCallback(
+    (value) => {
+      setSelectedTimezone(value);
+      if (value[0].label) updateTimeframe({ timezone: value[0].label });
+    },
+    [updateTimeframe, setSelectedTimezone]
+  );
+
+  const [startH, startM] = useMemo(() => timeframe.hours.start.split(':').map(Number), [timeframe]);
+  const [endH, endM] = useMemo(() => timeframe.hours.end.split(':').map(Number), [timeframe]);
+
+  return (
+    <>
+      <EuiSwitch
+        label={i18n.translate(
+          'xpack.triggersActionsUI.sections.actionTypeForm.ActionAlertsFilterTimeframeToggleLabel',
+          {
+            defaultMessage: 'If alert is generated during timeframe',
```

Comment:
>            defaultMessage: 'If alert is generated during timeframe',

This feels like an incomplete message. It makes me wonder what the end of the sentence is: ".... generate actions", "...suppress actions", "...wait to trigger actions"? If it's the first option, something like this is clearer IMO:

>           defaultMessage: 'Generate actions only in selected time window',

Reply:
@mdefazio are you okay with this change?

Reply:
Question about this toggle button: Normally, a toggle implies that you're turning something on or off. Would it make sense to label this toggle `Enable scheduled alert notifications` and then move the current default message to an informational tip? If you don't think it's necessary to include the verb "enable" because the act of enabling/disabling a toggle button is fairly intuitive, here's another option: `Generate scheduled alert notifications` 

If the conversation about naming the toggle has already been had and the decision was to go with a descriptive sentence instead, here's my suggestion:

>           defaultMessage: 'Send notifications in the selected time range',

Reply:
+1 to @nastasha-solomon's suggestion for more specific text for the toggle.

Reply:
I like `Send alert notifications within the selected time frame` when the toggle switch is enabled, but I feel like it's confusing when the switch is disabled. Maybe `within a time frame`?

Reply:
@jmikell821 
> I too, am confused about what this option does.

This option limits an action to only fire between the hours of X and Y, on Z days of the week. Not sure if "time frame" is the right wording on that, what are your thoughts?

Reply:
@Zacqary how about `Send alert notifications within the selected time frame only`? I think this implies that you're sending alerts during the selected schedule you choose. Another option is `Send alert notifications during the selected schedule only` but I like the former option better. I think when the option is not switched on, it'll clearly imply that notifications are _not_ sent. And of course, we'll be sure to indicate that in docs.

Reply:
Thanks for the ping! I like the suggestion.  Appreciate the extra eyes on this.  🙌